src/gui/modalMenu.cpp

```diff
@@ -40,7 +41,8 @@ GUIModalMenu::GUIModalMenu(gui::IGUIEnvironment* env, gui::IGUIElement* parent,
 		m_menumgr(menumgr),
 		m_remap_dbl_click(remap_dbl_click)
 {
-	m_gui_scale = std::max(g_settings->getFloat("gui_scaling"), 0.5f);
+	const double screen_dpi_scale = RenderingEngine::getDisplayDensity();
+	m_gui_scale = std::max(g_settings->getFloat("gui_scaling"), 0.5f) * screen_dpi_scale;
 #ifdef HAVE_TOUCHSCREENGUI
 	float d = RenderingEngine::getDisplayDensity();
```

Comment:
this is already done here - I'd suggest moving the getDisplayDensity out of the ifdef and adding an `#else`

Also, not sure why this is Android only and why the logic does *(1.1 - 0.3 * d + 0.2 * d * d)

Reply:
Good call I didn't see the ifdef below was doing almost the same thing